### ISSUE TITLE
Beacon chain panics on libp2p error

### ISSUE THREAD

**Issue by shayzluf** [prysmaticlabs/prysm#3686]
getting this error from libp2p
```
panic: runtime error: index out of range -1
goroutine 192 [running]:
github.com/libp2p/go-libp2p-peerstore/pstoremem.(*addrSegments).get(...)
    external/com_github_libp2p_go_libp2p_peerstore/pstoremem/addr_book.go:41
github.com/libp2p/go-libp2p-peerstore/pstoremem.(*memoryAddrBook).Addrs(0xc0003b4900, 0x0, 0x0, 0x0, 0x0, 0x0)
    external/com_github_libp2p_go_libp2p_peerstore/pstoremem/addr_book.go:236 +0x41c
github.com/libp2p/go-libp2p/p2p/host/routed.(*RoutedHost).Connect(0xc000301480, 0x5488580, 0xc0008fbec0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x40f5c9b, 0x5488540)
    external/com_github_libp2p_go_libp2p/p2p/host/routed/routed.go:62 +0xd8
github.com/libp2p/go-libp2p/p2p/host/routed.(*RoutedHost).NewStream(0xc000301480, 0x5488580, 0xc0008fbec0, 0x0, 0x0, 0xc0005cc4a0, 0x1, 0x1, 0x0, 0xc002cee280, ...)
    external/com_github_libp2p_go_libp2p/p2p/host/routed/routed.go:179 +0x1b0
github.com/prysmaticlabs/prysm/beacon-chain/p2p.(*Service).Send(0xc000178400, 0x5488580, 0xc0008fbec0, 0x50974a0, 0xc002cee280, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
    beacon-chain/p2p/sender.go:22 +0x2b6
github.com/prysmaticlabs/prysm/beacon-chain/sync/initial-sync.(*InitialSync).requestBlocks(0xc0000a0d50, 0x5488540, 0xc0000c6008, 0xc002cee280, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
    beacon-chain/sync/initial-sync/round_robin.go:193 +0x351
github.com/prysmaticlabs/prysm/beacon-chain/sync/initial-sync.(*InitialSync).roundRobinSync(0xc0000a0d50, 0x0, 0xed5224f57, 0x5fc7ca0, 0x1, 0xc000354608)
    beacon-chain/sync/initial-sync/round_robin.go:175 +0xa8e
github.com/prysmaticlabs/prysm/beacon-chain/sync/initial-sync.(*InitialSync).Start(0xc0000a0d50)
    beacon-chain/sync/initial-sync/service.go:107 +0x5a9
created by github.com/prysmaticlabs/prysm/shared.(*ServiceRegistry).StartAll
    shared/service_registry.go:44 +0x23e```

after seeing these lines during initial sync

```[2019-10-01 10:42:18]  INFO initial-sync: Processing block 1363/27641. Estimated 1h9m31s remaining. blocks per second=6.3 peers=6/0```

**Comment by prestonvanloon** [comment]
can you paste the whole error? I panicked on libp2p?

**Comment by shayzluf** [comment]
@prestonvanloon pasted it. thanks

**Comment by terencechain** [comment]
Is this still reproducible?

**Comment by prestonvanloon** [comment]
So this is when your only peer went away. I believe was recently fixed by @nisdas in #4130 or #4128
